feat(ui5-bar): allow role change

tsanislavgatev opened the pull request:
Introduces a new `accessibleRole` property to the `ui5-bar` component, allowing developers to customize the ARIA role of the bar for accessibility purposes.

By default, the role is set to "toolbar", but now it can be explicitly changed by the application.

fixes: https://github.com/SAP/ui5-webcomponents/issues/10990

unazko commented on packages/main/src/Bar.ts:
```diff
@@ -65,6 +67,16 @@ class Bar extends UI5Element {
+	 *
+	 */
+	@property()
+	accessibleRole: `${BarAccessibleRole}` = "Toolbar";
```
We'll need to provide recommendations to the users on how to use this new property.
In the openui5 project the logic is internal and no explanations were required.
Also in order to properly test this ourselves we could provide a dev sample that changes the number of interactive elements on runtime and therefore changes the role accordingly.

tsanislavgatev replied:
I've added additional documentation and example. But i've added a static example with just two bars explaining the both cases, because we don't provide dynamic change of the role, so such example wont be needed.

Stoev reviewed (commented):
Please consider adding short explanation:

feat(ui5-bar): allow role changes - maybe "role changes" is more explanatory?..
OR, eventually:
feat(ui5-bar): support for specific roles - just another suggestion...

And maybe consider adding some explanatory text, for example:
- Added new property `accessibleRole` to the `Bar` class.
- `Bar` component now supports custom roles for specific use cases.
- Updated the `accInfo` method to include the `effectiveRole`.

tsanislavgatev commented:
> Please consider adding short explanation:
> 
> feat(ui5-bar): allow role changes - maybe "role changes" is more explanatory?.. OR, eventually: feat(ui5-bar): support for specific roles - just another suggestion...
> 
> And maybe consider adding some explanatory text, for example:
> 
> * Added new property `accessibleRole` to the `Bar` class.
> * `Bar` component now supports custom roles for specific use cases.
> * Updated the `accInfo` method to include the `effectiveRole`.

Hi @Stoev, 

I've just added the documentation. The version you've reviewed is pre-documentation. Can you please review it now? I've added API documentation with example when and how it should be used. Also an additional sample was added showing both cases of the bar usage.

Stoev commented on packages/main/src/Bar.ts:
```diff
@@ -65,6 +67,23 @@ class Bar extends UI5Element {
+	 *
+	 * - Set accessibleRole to "toolbar" only when the component contains two or more active, interactive elements (such as buttons, links, or input fields) within the bar.
+	 *
+	 * - If there is only one or no active element, it is recommended to avoid using the toolbar role, as it implies a grouping of multiple interactive controls.
```
Please consider:
"toolbar"

Stoev reviewed (commented):
Just minor recommendations:

Introduces a new `accessibleRole` property to the `ui5-bar` component, allowing developers to customize the ARIA role of the bar for accessibility purposes.

By default, the role is set to "toolbar", but now it can be explicitly changed by the application.
